java/core/src/main/java/co/worklytics/psoxy/impl/SanitizerImpl.java

```diff
@@ -388,4 +390,38 @@ public PseudonymizedIdentity pseudonymize(@NonNull Number value) {
         return pseudonymize((Object) value);
     }
 
+    List<Pair<Pattern, List<JsonPath>>>  getCompiledPseudonymizations() {
+        if (compiledPseudonymizations == null) {
+            synchronized ($writeLock){
```

Comment:
ideally you should re-check the null, in case it's been done while getting the lock
See double-checked locking pattern
https://www.baeldung.com/java-singleton-double-checked-locking
https://www.cs.umd.edu/~pugh/java/memoryModel/DoubleCheckedLocking.html